lib/semian/circuit_breaker.rb

```diff
@@ -43,70 +44,77 @@ def mark_failed(_error)
 
     def mark_success
       return unless half_open?
-      @successes += 1
+      @successes.increment
       close if success_threshold_reached?
     end
 
     def reset
-      @errors = []
-      @successes = 0
+      @errors.clear
+      @successes.value = 0
```

Comment:
Probably should have included this as well: 
```ruby
# From the Enum
def value=(sym)
  @integer.value = @sym_to_num.fetch(sym)
end
```
They don't inherit from each other anymore, it's a `Forwardable`, and I did that to get rid of that diamond dependency. 

Although, if we get rid of it like this, the hash lookup will only be in the `SysV` so there won't be code sharing, and there won't be a diamond dependency issue, and no `def_delegators` needed. What about that? `SysV::Enum` will have to inherit from `SysV::Integer`.